### ISSUE TITLE
Base name in FileAnalyzerFactory find not working properly

### ISSUE THREAD

**Issue by gtoph** [oracle/opengrok#1188]
It appeared at least on windows that getting the base name is not properly handled, particularly when the file doesn't have an extension.  I believe the issue is with path slash style.  Don't have a ton of knowledge of opengrok source yet, but doing this seems to at least work for me now in that it returns the correct analyzer:
```java
**final String lpath = path.replaceAll("[/\\\\]", Matcher.quoteReplacement(File.separator));**
        if (((i = lpath.lastIndexOf(File.separatorChar)) > 0)
                && (i + 1 < lpath.length())) {
            path = lpath.substring(i + 1);
        }
```

**Comment by tarzanek** [comment]
hmm, I develop on Win 10 and never hit this
can you provide EXACT steps to reproduce?
(ideally with source path structure, ev. with the repo you're trying to index)

**Comment by gtoph** [comment]
Sorry should have probably spec'ed that before.  So some quick info:

1. On windows 10 as well
2. Added a 'sconscript' analyzer (basically a copy of python for now) and added "SConscript" to the specific FILE_NAMES list
3. Working with perforce repo's, though for this test, I simply have a few files in a typical directory layout
4. Command line: 
-v -S -H -a on -W C:\Projects\Test\test.xml -c C:\Projects\Test\ctags58\ctags.exe -d C:\Projects\Test\TestData -P -O on -s C:\Projects\Test\testproject

This little snipet never seems to run, so path == file :

       if (((i = path.lastIndexOf(File.separatorChar)) > 0)
                && (i + 1 < path.length())) {
            path = path.substring(i + 1);
        }

In a file with an extension this will work fine as there will be a dotpos and prefix/suffix will match
(i.e. "/test/a/b/c/api.h")

Basically it looks like paths are coming in with '/' but File.separatorChar is '\', so it'll never match.

So in my particular case, I have a file "/test/a/b/SConscript" (no extension).
'path' is still is the same as file, and dotpos = -1, so we fall through to:

return FILE_NAMES.get(path.toUpperCase(Locale.getDefault()));

Since path is the full path and not just the file name, this fails and hence we don't associate the correct analyzer.

So first doing 
path.replaceAll("[/\\\\]", Matcher.quoteReplacement(File.separator));

obviously flips all the slashes to the same way that lastIndexOf is looking for, so now we actually run that if block and 'path' is now just "SConscript" (or whatever actual filename), and FILE_NAMES can get a match.

Hope that makes sense.

**Comment by tarzanek** [comment]
I really need to share my bat file I use to run indexer ... I think I even shared it in one of older issues ...

**Comment by tarzanek** [comment]
in #926 actually I have the command I use

**Comment by tarzanek** [comment]
and in #1091

**Comment by vladak** [comment]
Cannot reproduce this on Windows. Either the problem went away via some change or there is very specific code path that makes the `AnalyzerGuru#find(String)` to be called with non-native path separator. Feel free to reopen with more detail.
